File: processor/annotations/src/main/java/com/google/appinventor/components/annotations/SimpleFunction.java
package com.google.appinventor.components.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation to mark Simple functions.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface SimpleFunction {
    /**
     * If non-empty, description to use in user-level documentation in place of Javadoc, which is meant for developers.
     *
     * @return the description of the function
     */
    String description() default "";
}

File: processor/annotations/src/main/kotlin/com/google/appinventor/components/annotations/SimpleFunction.kt
package com.google.appinventor.components.annotations

/**
 * Annotation to mark Simple functions.
 *
 * Note that the Simple compiler will only recognize Java methods marked
 * with this annotation. All other methods will be ignored.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class SimpleFunction(
    /**
     * If non-empty, description to use in user-level documentation in place of
     * Javadoc, which is meant for developers.
     */
    val description: String = ""
)
